src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,812 @@
+
+const LocomotorTemplate *LocomotorStore::Find_Locomotor_Template(NameKeyType namekey) const
+{
+    if (namekey == NAMEKEY_INVALID) {
+        return nullptr;
+    }
+
+    auto t = m_locomotorTemplates.find(namekey);
+
+    if (t != m_locomotorTemplates.end()) {
+        return (*t).second;
+    } else {
+        return nullptr;
+    }
+}
+
+void LocomotorStore::Reset()
+{
+    auto it = m_locomotorTemplates.begin();
+
+    for (;;) {
+        if (!(it != m_locomotorTemplates.end())) {
+            break;
+        }
+
+        if (it->second->Delete_Overrides() == nullptr) {
+            m_locomotorTemplates.erase(it);
+        } else {
+            ++it;
+        }
+    }
+}
+
+Locomotor *LocomotorStore::New_Locomotor(const LocomotorTemplate *tmpl)
+{
+    return new Locomotor(tmpl);
+}
+
+LocomotorTemplate *LocomotorStore::New_Override(LocomotorTemplate *tmpl)
+{
+    if (tmpl == nullptr) {
+        return nullptr;
+    }
+
+    LocomotorTemplate *t = new LocomotorTemplate(*tmpl);
+    tmpl->Set_Next(t);
+    t->Set_Is_Allocated();
+    return t;
+}
+
+void LocomotorStore::Parse_Locomotor_Template_Definition(INI *ini)
+{
+    if (g_theLocomotorStore == nullptr) {
+        throw CODE_06;
+    }
+
+    bool found = false;
+    const char *name = ini->Get_Next_Token();
+    NameKeyType key = g_theNameKeyGenerator->Name_To_Key(name);
+
+    LocomotorTemplate *t = g_theLocomotorStore->Find_Locomotor_Template(key);
+    if (t != nullptr) {
+        if (ini->Get_Load_Type() == INILoadType::INI_LOAD_CREATE_OVERRIDES) {
+            t = g_theLocomotorStore->New_Override(static_cast<LocomotorTemplate *>(t->Friend_Get_Final_Override()));
+        }
+
+        found = true;
+    } else {
+        t = NEW_POOL_OBJ(LocomotorTemplate);
+
+        if (ini->Get_Load_Type() == INILoadType::INI_LOAD_CREATE_OVERRIDES) {
+            t->Set_Is_Allocated();
+        }
+    }
+
+    t->Set_Name(name);
+    ini->Init_From_INI(t, LocomotorTemplate::Get_Field_Parse());
+    t->Validate();
+
+    if (!found) {
+        g_theLocomotorStore->m_locomotorTemplates[key] = t;
+    }
+}
+
+Locomotor::Locomotor(const LocomotorTemplate *tmpl) :
+    m_brakingFactor(1.0f),
+    m_maxLift(99999.0f),
+    m_maxSpeed(99999.0f),
+    m_maxAccel(99999.0f),
+    m_maxBreaking(99999.0f),
+    m_maxTurnRate(99999.0f),
+    m_flags(0)
+{
+    m_template = tmpl;
+    m_closeEnoughDist = m_template->m_closeEnoughDist;
+    Set_Flag(CLOSE_ENOUGH_DIST_3D, m_template->m_closeEnoughDist3D);
+    m_preferredHeight = m_template->m_preferredHeight;
+    m_preferredHeightDamping = m_template->m_preferredHeightDamping;
+    m_wanderAngle = Get_Logic_Random_Value_Real(DEG_TO_RADF(-30), DEG_TO_RADF(30));
+    m_wanderLength = Get_Logic_Random_Value_Real(0.80000001f, 1.2f);
+    Set_Flag(WANDER_DIRECTION, (Get_Logic_Random_Value(0, 1) != 0));
+    m_moveFrame = g_theGameLogic->Get_Frame() + 2.5f * 30.0f;
+}
+
+Locomotor::Locomotor(const Locomotor &that) :
+    m_brakingFactor(that.m_brakingFactor),
```

Comment:
`m_template` and `m_moveFrame` appear to be missing here.

Reply:
m_moveFrame is explicitly not being copied here. Fixed the other thing.